### PR TITLE
[Docs] Japanese translation of feature_rawhid.md

### PR DESCRIPTION
**Pull request by umi-umi**
## Description

Add Japanese translation of docs/feature_rawhid.md.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mtei** [comment]
@shelaf @s-show @hsgw @satt99 @takashiski @yoshihara よかったらレビューお願いします。

File: docs/ja/feature_rawhid.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+* Python: [pyhidapi](https://pypi.org/project/hid/)。
+
+これは完全なクロスプラットフォームのリストではありませんが、最初に始めるのに十分なはずです。raw HID を使うための特別な要件は無いため、どの HID ライブラリでも動作するはずです。
+
+これで、キーボードへの HID インタフェースを開くために必要な4つの情報全てが揃いました。必要なのは、ライブラリの利用可能な関数を使って ID パラメータを使ってデバイスを開くことだけです。
+
+Vendor ID と Product ID はデバイスを開くために実際には必要ないことに注意してください。それらは接続した多くの HID デバイスから特定のデバイスをフィルターするためだけに使われます。多くのライブラリでは、代わりに製品名と製造元名を使ってデバイスを開くオプションがあります。`node-hid` が代表的な例です。このことは、USB ハブが組み込まれているデバイスや、同じ製品名または同じ製造元の複数のインタフェースがある追加の HID インタフェースで問題になります。Product ID と Vendor ID を合わせると単一のインタフェースが一意の宛先になるため、この問題を防げます。したがって、ライブラリで必要が無い場合でも、この問題を防ぐためにそれらを使うことをお勧めします。
+ただし、Vendor ID や Product ID と異なり、Usage Page と Usage は通信を成功させるために必要です。
+
+言うまでもなく、使っているライブラリに関係なく、終了したらインタフェースを必ずインタフェースを閉じる必要があります。オペレーティングシステムと特定の環境によっては、明示的に接続が閉じられていない場合、後で他のクライアントまたは同じクライアントの他のインスタンスに再接続する時に問題が発生する可能性があります。
```
Comment (hsgw) [inline docs/ja/feature_rawhid.md:74]:
```suggestion
言うまでもなく、使っているライブラリに関係なく、終了したらインタフェースを必ず閉じる必要があります。オペレーティングシステムと特定の環境によっては、明示的に接続が閉じられていない場合、後で他のクライアントまたは同じクライアントの他のインスタンスに接続しなおした時に問題が発生する可能性があります。
```

**Review by hsgw — CHANGES_REQUESTED** [review]

File: docs/ja/feature_rawhid.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+* Node: [node-hid](https://github.com/node-hid/node-hid)。
+* C: [hidapi](https://github.com/libusb/hidapi)。
+* Java: [purejavahidapi](https://github.com/nyholku/purejavahidapi) と [hid4java](https://github.com/gary-rowe/hid4java)。
+* Python: [pyhidapi](https://pypi.org/project/hid/)。
+
+これは完全なクロスプラットフォームのリストではありませんが、最初に始めるのに十分なはずです。raw HID を使うための特別な要件は無いため、どの HID ライブラリでも動作するはずです。
+
+これで、キーボードへの HID インタフェースを開くために必要な4つの情報全てが揃いました。必要なのは、ライブラリの利用可能な関数を使って ID パラメータを使ってデバイスを開くことだけです。
+
+Vendor ID と Product ID はデバイスを開くために実際には必要ないことに注意してください。それらは接続した多くの HID デバイスから特定のデバイスをフィルターするためだけに使われます。多くのライブラリでは、代わりに製品名と製造元名を使ってデバイスを開くオプションがあります。`node-hid` が代表的な例です。これは USB ハブが組み込まれているデバイスや、同じ製品名または同じ製造元の複数のインタフェースがある特別な HID インタフェースで問題になります。Product ID と Vendor ID を合わせると単一のインタフェースがユニークな宛先になるため、この問題を防げます。したがって、ライブラリで必要が無い場合でも、この問題を防ぐためにそれらを使うことをお勧めします。
```
Comment (shelaf) [inline docs/ja/feature_rawhid.md:71]: designationはdestinationと間違えて翻訳されている？
```suggestion
Vendor ID と Product ID はデバイスを開くために実際には必要ないことに注意してください。それらは接続した多くの HID デバイスから特定のデバイスをフィルターするためだけに使われます。多くのライブラリでは、代わりに製品名と製造元名を使ってデバイスを開くオプションがあります。`node-hid` が代表的な例です。これは USB ハブが組み込まれているデバイスや、同じ製品名または同じ製造元の複数のインタフェースがある特別な HID インタフェースで問題になります。Product ID と Vendor ID を合わせると単一のインタフェースの固有名を作成できるため、この問題を防げます。したがって、ライブラリで必要が無い場合でも、この問題を防ぐためにそれらを使うことをお勧めします。
```

Comment (umi-umi) [reply docs/ja/feature_rawhid.md]: ご指摘ありがとうございます。
まさに、その読み間違いをしていました。

**Review by shelaf — APPROVED** [review]

File: docs/ja/feature_rawhid.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+`keymap.c` に `"raw_hid.h"` を include し、以下を実装します:
+
+```C
+void raw_hid_receive(uint8_t *data, uint8_t length) {
+    // ここにコードを書きます。data はホストから受信したパケットです。
+}
+```
+
+`"raw_hid.h"` ヘッダは、キーボードからホストにパケットを送信できる `void raw_hid_send(uint8_t *data, uint8_t length);` も宣言します。例として、全てのデータをホストに返すことでホストアプリケーションを構築する時にデバッグをするために使うこともできます。
```
Comment (s-show) [inline docs/ja/feature_rawhid.md:29]: 2番目の文の長さを考えると、句読点がもう1つ必要かなと思いますので、次の訳はどうでしょうか？
```suggestion
`"raw_hid.h"` ヘッダは、キーボードからホストにパケットを送信できる `void raw_hid_send(uint8_t *data, uint8_t length);` も宣言します。例として、全てのデータをホストに返すことで、ホストアプリケーションを構築する時のデバッグに使うこともできます。
```

**Review by s-show — APPROVED** [review]

**Review by hsgw — APPROVED** [review]

**Review by mtei — APPROVED** [review]

**Comment by mtei** [comment]
I think this is ready to merge.

umi\-umi さん、その他のレビュアーの皆さん、大変お待たししました。_o_

**Comment by mtei** [comment]
Thanks!

**Pull request merged**